File: TicTacToeGame/ObjectModel/PlayerTurn.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace Edu.Schwabe.TicTacToe.ObjectModel
{
  [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
  public class PlayerTurn
  {
    public SpecificCellOwner CellOwner { get; set; }

    public GameView GameView { get; } = new GameView();

    public GameActionHistory ActionHistory { get; } = new GameActionHistory();
  }
}

File: tic-tac-toe-client/src/meta-model/PlayerTurn.ts
import { GameActionHistory } from './GameActionHistory';
import { GameView } from './GameView';
import { SpecificCellOwner } from './CellOwner';

export interface PlayerTurn {
  cellOwner: Readonly<SpecificCellOwner>;
  gameView: Readonly<GameView>;
  actionHistory?: Readonly<GameActionHistory>;
}
